internal/tuf/v02/root.go

```diff
@@ -757,3 +758,112 @@ func (o *OtherRepository) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// Hooks defines the schema that groups hooks by stage.
+type Hooks struct {
```

Comment:
Do we need Hooks to be a wrapper type? It could just be `map[tuf.HookStage]map[string]*Hook` right? Or even `map[tuf.HookStage][]*Hook` since each hook also has a notion of its name.

Reply:
This way, we can also look up directly using a tuf.HookStage const without having to rely on `.String()` at lookup time. `tuf.HookStage` may need a custom json marshal / unmarshal though, if we want it to be readable in the metadata.